lib/vanagon/engine/hardware.rb

```diff
@@ -0,0 +1,72 @@
+require 'vanagon/engine/base'
+require 'json'
+require 'lock_manager'
+
+LOCK_MANAGER_HOST = ENV['LOCK_MANAGER_HOST'] || 'redis'
+LOCK_MANAGER_PORT = ENV['LOCK_MANAGER_PORT'] || 6379
+VANAGON_LOCK_USER = "#{ENV['USER']}"
+
+
+class Vanagon
+  class Engine
+    # Class to use when building on a hardware device (e.g. AIX, Switch, etc)
+    #
+    class Hardware < Base
+
+      # This method is used to obtain a vm to build upon
+      # For the base class we just return the target that was passed in
+      def select_target
+        if @build_hosts.length == 1
+          @target = single_node_lock(@build_hosts[0])
+        else
+          @target = multi_node_lock(@build_hosts)
+        end
+      end
+
+      # Go through the lock process for a single node
+      def single_node_lock(host)
```

Comment:
This seems like an unnecessary special case. Even for 1 node you can still do `hosts.each`. In the multi_node_lock you can put in a timeout on a polling lock and then there would be just one locking method.